src/SDKs/Batch/Support/FileStaging/Batch.FileStaging/FileStagingLinkedSources.cs

```diff
@@ -44,20 +36,10 @@ private static string MakeDefaultNamePlusNamingFragment(string namingFragment)
             return newName;
         }
 
-        // lock used to ensure only one name is created at a time.
-        private static object _lockForContainerNaming = new object();
-
         internal static string ConstructDefaultName(string namingFragment)
         {
-            lock (_lockForContainerNaming)
-            {
-                Thread.Sleep(30);  // make sure no two names are identical
-
-                string uniqueLetsHope = DateTime.UtcNow.ToString("yyyy-MM-dd-HH-mm-ss-fff");
-                string defContainerName = MakeDefaultNamePlusNamingFragment(namingFragment) + uniqueLetsHope;
-
-                return defContainerName;
-            }
+            string containerName = MakeDefaultNamePlusNamingFragment(namingFragment) + Guid.NewGuid().ToString("D");
```

Comment:
I updated this to use a time (for ordering) _and_ a GUID for uniqueness at the end. Plus added some max length truncation where we will truncate the container name if it's too long.

That saves us the `Thread` (which we can't use) but I think maintains the original behavior.